strip=debuginfo is passed to the linker as `--strip-debug` which
our ld(1) does not want to deal with

diff -wpruN '--exclude=*.orig' a~/src/tools/compiletest/src/runtest.rs a/src/tools/compiletest/src/runtest.rs
--- a~/src/tools/compiletest/src/runtest.rs	1970-01-01 00:00:00
+++ a/src/tools/compiletest/src/runtest.rs	1970-01-01 00:00:00
@@ -1952,7 +1952,7 @@ impl<'test> TestCx<'test> {
                 rustc.arg("-Zui-testing");
                 rustc.arg("-Zdeduplicate-diagnostics=no");
                 // FIXME: use this for other modes too, for perf?
-                rustc.arg("-Cstrip=debuginfo");
+                // rustc.arg("-Cstrip=debuginfo");
             }
             MirOpt => {
                 rustc.args(&[
